src/components/BigNumber/BigNumber.tsx

```diff
@@ -0,0 +1,273 @@
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { FC, cloneElement, useRef } from 'react';
+
+import { BIG_NUMBER_ASPECT_RATIO } from '@constants';
+import { Line } from '@rsc';
+import { sanitizeBigNumberChildren } from '@utils';
+import { RscChart } from 'RscChart';
+import { BigNumberMethod, BigNumberProps, ChartData, IconElement, LineElement, LineProps, Orientation } from 'types';
+import { getLocale } from 'utils/locale';
+import { v4 as uuid } from 'uuid';
+import { View as VegaView } from 'vega';
+
+import { Flex, Grid, View } from '@adobe/react-spectrum';
+
+import './BigNumber.css';
+import { getFormattedString } from './bigNumberFormatUtils';
+
+const BigNumber: FC<BigNumberProps> = ({
+	dataKey,
+	label,
+	numberFormat,
+	children,
+	icon,
+	orientation,
+	rscChartProps,
+	numberType = 'linear',
+	method = 'last',
+}) => {
+	const chartId = useRef<string>(`rsc-${uuid()}`);
+	const chartView = useRef<VegaView>();
+
+	rscChartProps = rscChartProps ?? {
+		data: [],
+		chartId,
+		chartView,
+		chartWidth: 200,
+	};
+
+	const { chartWidth, height, locale, data, ...rscChartRemain } = rscChartProps;
+	const aspectRatio = BIG_NUMBER_ASPECT_RATIO;
+	// based on Chart.tsx checks, data will always be defined and have a length greater than 0.
+	const bigNumberValue = getBigNumberValue(data, dataKey, method);
+
+	const numberLocale = getLocale(locale).number;
+
+	const formattedValue = getFormattedString(bigNumberValue, numberType, numberLocale, numberFormat);
+
+	const lineElements = sanitizeBigNumberChildren(children);
+	const lineProps = lineElements.length > 0 ? (lineElements[0] as LineElement).props : undefined;
+
+	let cWidth: number, cHeight: number, generalJustify: 'center' | 'start', padding: number;
+
+	if (orientation == 'vertical') {
+		padding = 0;
+		cHeight = height ? height / 2.25 : chartWidth / aspectRatio;
+		cWidth = height ? cHeight * aspectRatio : chartWidth;
+		generalJustify = 'center';
+	} else {
+		if (height && height < chartWidth / (1.75 * aspectRatio)) {
+			cHeight = height;
+			cWidth = height * aspectRatio;
+		} else {
+			cWidth = chartWidth / 1.75;
+			cHeight = cWidth / aspectRatio;
+		}
+		padding = 10;
+		generalJustify = 'start';
+	}
+
+	const { areas, columns, iconAlign, labelAlign, iconJustify, iconSize, displayCombo } = getGridProperties(
+		icon,
+		orientation,
+		height,
+		chartWidth,
+		lineProps
+	);
+	return (
+		<div tabIndex={0} className="big-number-container">
```

Comment:
Let's just drop the tabIndex for now. We don't have accessibility support yet, so we can come back to this when we do a larger push for it in the future.